pkg/eventpersisterdwh/persister/goal_event.go

```diff
@@ -319,28 +323,39 @@ func (w *goalEvtWriter) listExperiments(
 	exp, err, _ := w.flightgroup.Do(
 		fmt.Sprintf("%s:%s", environmentNamespace, "listExperiments"),
 		func() (interface{}, error) {
-			experiments := []*exproto.Experiment{}
-			cursor := ""
-			for {
-				resp, err := w.experimentClient.ListExperiments(ctx, &exproto.ListExperimentsRequest{
-					PageSize:             listRequestSize,
-					Cursor:               cursor,
-					EnvironmentNamespace: environmentNamespace,
-					Statuses: []exproto.Experiment_Status{
-						exproto.Experiment_RUNNING,
-						exproto.Experiment_STOPPED,
-					},
-				})
-				if err != nil {
-					return nil, err
-				}
-				experiments = append(experiments, resp.Experiments...)
-				experimentSize := len(resp.Experiments)
-				if experimentSize == 0 || experimentSize < listRequestSize {
-					return experiments, nil
-				}
-				cursor = resp.Cursor
+			// Get the experiment cache
+			expList, err := w.cache.Get(environmentNamespace)
+			if err == nil {
+				return expList.Experiments, nil
+			}
+			w.logger.Info(
```

Comment:
@masaaania, maybe it will be better if we remove this log because it will log every minute.